e2e/cypress/support/api_commands.js

```diff
@@ -56,6 +56,54 @@ Cypress.Commands.add('apiEmailTest', () => {
 // https://api.mattermost.com/#tag/posts
 // *****************************************************************************
 
+/**
+* Creates a post directly via API
+* This API assume that the user is logged in and has cookie to access
+* @param {String} channelId - Where to post
+* @param {String} message - What to post
+* @param {String} rootId - Parent post ID. Set to "" to avoid nesting
+* @param {Object} props - Post props
+* @param {String} token - Optional token to use for auth. If not provided - posts as current user
+*/
+Cypress.Commands.add('apiCreatePost', (channelId, message, rootId, props, token = '', failOnStatusCode = true) => {
```

Comment:
I understand that this is a generic API command for posting message via API but like to mention that there's an existing `cy.postBotMessage` command that do specifically for posting bot message. Maybe just need to [workaround to allow failing status](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/support/task_commands.js#L15). It acts like an external request where it doesn't change the current session. So no need to logout and login the existing user.